### ISSUE TITLE
Question about PubMed Dataset (Potential Data Leakage)

### ISSUE THREAD

**Issue by rthapa84** [UCSC-VLAA/m1#4]
Hello, 

I was looking at your dataset (UCSC-VLAA/m23k-tokenized) and I noticed that there is overlap with qiaojin/PubMedQA (pqa_labeled). I was wondering what dataset are you using for evaluation for PubMedQA?

**Comment by xk-huang** [comment]
Hi @rthapa84 

Thanks for your interest in our work!

1. Our training set have **no data contamination** with the eval set.
Decontamination is conducted in `python src/collect_data/merge_data.py` and `python src/distill_data/decontaminate_eval.py`.

2. PubMedQA (qiaojin/PubMedQA (pqa_labeled)) has 1K samples, but they are not split into train/test, meaning everyone has to split by their own, and there is no de facto approach. Even in the original paper of PubMedQA, the authors arranged their 1K samples into 10 different (10-fold) splits of train/test sets of 50:50. For the test split of PubMedQA, we use those from HuaTuoGPT-o1, so we decontaminate our train split against that of HuaTuoGPT-o1 in `python src/distill_data/decontaminate_eval.py`.

**Comment by rthapa84** [comment]
Hey @xk-huang, thank you for your response. I understand your explanation, however, I am still not fully convinced. I checked the eval files you provided here UCSC-VLAA/m1_eval_data, and i compared it against (UCSC-VLAA/m23k-tokenized) and qiaojin/PubMedQA (pqa_labeled). Here are the overlaps I found: https://drive.google.com/file/d/1AscbFVRzUR9wfWd2ca4RnOTIq5cAm1nT/view?usp=drive_link.

**Comment by xk-huang** [comment]
Hi @rthapa84,

Thanks for providing the comparison! I now see that there are 28 overlapping samples.

At the start, to ensure that our PubMedQA training samples do not overlap with the test set, we perform decontamination by running `python src/distill_data/decontaminate_eval.py`.

You can track the number of training samples throughout this process here: [PubMedQA Data Analysis Notebook](https://github.com/UCSC-VLAA/m1/blob/main/src/analyse/stats_data_source_domain.ipynb).

We begin with a random split of 500 training samples:
1. After filtering out easy samples: 116 remain
2. After filtering out incorrect responses: 39 remain
3. After decontamination against 500 test samples from [HuatuoGPT-o1](https://github.com/FreedomIntelligence/HuatuoGPT-o1): 28 remain

However, it appears the decontamination script missed 28 overlapping samples. We will fix this issue by first removing these 28 samples from the training set, and then retraining the model and reporting the final performance.

**Comment by rthapa84** [comment]
@xk-huang Thank you for addressing this — I really appreciate it. I think, as a community, we should try to avoid using the test set (even though I see some other reasoning papers doing the same to improve benchmark results) as much as possible. It makes it much harder for others to compare fairly, and it forces everyone to follow a specific test set for each new paper.

Anyway, thank you again for addressing this. I’m looking forward to the update.
